src/main/java/uk/ac/cam/cl/dtg/segue/configuration/SegueGuiceConfigurationModule.java

```diff
@@ -1011,18 +1010,18 @@ private static SegueJobService getSegueJobService(final PropertiesLoader propert
                 configuredScheduledJobs.add(eventFeedbackEmail);
             }
 
-            segueJobService = new SegueJobService(configuredScheduledJobs, database);
-
-            // Simply removing the following jobs from the configuredScheduledJobs wouldn't remove them from the qrtz_job_details
-            // page so they need properly removing.
+            // Simply removing jobs from configuredScheduledJobs won't de-register them if they
+            // are currently configured, so the constructor takes a list of jobs to remove too.
+            List<SegueScheduledJob> scheduledJobsToRemove = new ArrayList<>();
             if (mailjetKey == null && mailjetSecret == null) {
-                segueJobService.removeScheduleJob(syncMailjetUsers);
+                scheduledJobsToRemove.add(syncMailjetUsers);
             }
 
             if (!eventPrePostEmailsEnabled) {
-                segueJobService.removeScheduleJob(eventReminderEmail);
-                segueJobService.removeScheduleJob(eventFeedbackEmail);
+                scheduledJobsToRemove.add(eventReminderEmail);
+                scheduledJobsToRemove.add(eventFeedbackEmail);
             }
+            segueJobService = new SegueJobService(database, configuredScheduledJobs, scheduledJobsToRemove);
```

Comment:
Just a thought: It's a bit annoying that Quartz not working meant that the API didn't work.
Should we have a try catch around this rather than throwing the SchedulerException?
In my head they are separate services that should be monitored separately but that does increase the complexity of the architecture a little.

Reply:
It must be said that no other provider in the GuiceConfigurationModule looks to throw a checked exception. Both Postgres and ElasticSearch throw unchecked exceptions, which have the same end effect.

I suppose we could catch and return null. And UptimeRobot does monitor the Quartz status, so we probably would notice in that case. We don't explicitly use the provider method, only via `injector.getInstance(SegueJobService.class);`, and so returning `null` (which other providers do when they fail to init) seems safe enough.

OTOH, Quartz is now critical to the normal function of the website, so perhaps failing hard and fast is safer overall?

Reply:
Let's leave it as is for now 👍